app/controllers/application_controller.rb

```diff
@@ -188,13 +188,23 @@ def catch_ember_cli_hijack
   rescue_from RateLimiter::LimitExceeded do |e|
     retry_time_in_seconds = e&.available_in
 
+    response_headers = {
+      'Retry-After': retry_time_in_seconds.to_s
+    }
+
+    if e&.error_code
+      response_headers.merge!(
+        'Discourse-Rate-Limit-Error-Code': e.error_code
+      )
```

Comment:
```suggestion
      response_headers['Discourse-Rate-Limit-Error-Code'] = e.error_code
```

This seems odd because we can just alter the existing hash instead of having to create a new hash to merge?